### PR TITLE
Implement personal_ecRecover

### PR DESCRIPTION
**Pull request by bbondy**
Implements personal_ecRecover and makes it compatible with MM.
This is used to recover the address from a message and signature that was used to sign the message.
Since this has nothing to do with permissioned accounts the keyring can be locked.

Resolves https://github.com/brave/brave-browser/issues/19621

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/common/eth_request_helper_unittest.cc [hunk 1]
```diff
@@ -316,8 +316,71 @@ TEST(EthResponseHelperUnitTest, ParsePersonalSignParams) {
   EXPECT_FALSE(ParsePersonalSignParams(json, nullptr, nullptr));
   EXPECT_FALSE(
       ParsePersonalSignParams("{\"params\":[{}]}", &address, &message));
```
Comment (darkdh) [inline components/brave_wallet/common/eth_request_helper_unittest.cc:318]: we should also add `ParsePersonalEcRecoverParams` to `EthRequestHelperUnitTest.CommonParseErrors`

File: components/brave_wallet/browser/hd_keyring.cc [hunk 2]
```diff
@@ -157,6 +159,59 @@ std::vector<uint8_t> HDKeyring::SignMessage(const std::string& address,
...
+  signature_only.pop_back();
+
+  // Get the 32 byte message hash
+  std::vector<uint8_t> hash;
+  std::string prefix("\x19");
+  prefix += std::string("Ethereum Signed Message:\n" +
+                        base::NumberToString(message.size()));
+  std::vector<uint8_t> hash_input(prefix.begin(), prefix.end());
+  hash_input.insert(hash_input.end(), message.begin(), message.end());
+  hash = KeccakHash(hash_input);
```
Comment (darkdh) [inline components/brave_wallet/browser/hd_keyring.cc:189]: We should put this into an internal function to be shared with `HDKeyring::SignMessage`

File: components/brave_wallet/common/eth_request_helper_unittest.cc (hunk 1 above)
Comment (bbondy) [reply components/brave_wallet/common/eth_request_helper_unittest.cc:321]: Fixed in rebase https://github.com/brave/brave-core/pull/11590/commits/83d9229f9bd1c95362238c415c7c02f7f6f6c868

File: components/brave_wallet/browser/hd_keyring.cc (hunk 2 above)
Comment (bbondy) [reply components/brave_wallet/browser/hd_keyring.cc]: Fixed in rebase https://github.com/brave/brave-core/pull/11590/commits/83d9229f9bd1c95362238c415c7c02f7f6f6c868
See `GetMessageHash`

**Review by darkdh — APPROVED** [review]

**Pull request merged**